packages/angular/src/tracing.ts

```diff
@@ -0,0 +1,208 @@
+// tslint:disable:max-classes-per-file
+
+import { AfterViewInit, Directive, Injectable, Input, OnInit } from '@angular/core';
+import { Event, NavigationEnd, NavigationStart, Router } from '@angular/router';
+import { getCurrentHub } from '@sentry/browser';
+import { Span, Transaction, TransactionContext } from '@sentry/types';
+import { logger, timestampWithMs } from '@sentry/utils';
+import { Observable } from 'rxjs';
+import { filter, tap } from 'rxjs/operators';
+
+let instrumentationInitialized: boolean;
+let stashedStartTransaction: (context: TransactionContext) => Transaction | undefined;
+let stashedStartTransactionOnLocationChange: boolean;
+
+/**
+ * Creates routing instrumentation for Angular Router.
+ */
+export function routingInstrumentation(
+  startTransaction: (context: TransactionContext) => Transaction | undefined,
+  startTransactionOnPageLoad: boolean = true,
+  startTransactionOnLocationChange: boolean = true,
+): void {
+  instrumentationInitialized = true;
+  stashedStartTransaction = startTransaction;
+  stashedStartTransactionOnLocationChange = startTransactionOnLocationChange;
+
+  if (startTransactionOnPageLoad) {
+    startTransaction({
+      name: window.location.pathname,
+      op: 'pageload',
+    });
+  }
+}
+
+/**
+ * Grabs active transaction off scope
+ */
+export function getActiveTransaction(): Transaction | undefined {
+  const currentHub = getCurrentHub();
+
+  if (currentHub) {
+    const scope = currentHub.getScope();
+    if (scope) {
+      return scope.getTransaction();
+    }
+  }
+
+  return undefined;
+}
+
+/**
+ * Angular's Service responsible for hooking into Angular Router and tracking current navigation process.
+ * Creates a new transaction for every route change and measures a duration of routing process.
+ */
+@Injectable({ providedIn: 'root' })
+export class TraceService {
+  private routingSpan?: Span;
+
+  public constructor(private readonly router: Router) {
+    this.navStart$.subscribe();
+    this.navEnd$.subscribe();
+  }
+
+  public navStart$: Observable<Event> = this.router.events.pipe(
+    filter(event => event instanceof NavigationStart),
+    tap(event => {
+      if (!instrumentationInitialized) {
+        logger.error('Angular integration has tracing enabled, but Tracing integration is not configured');
+        return;
+      }
+
+      const navigationEvent = event as NavigationStart;
+      let activeTransaction = getActiveTransaction();
+
+      if (!activeTransaction && stashedStartTransactionOnLocationChange) {
+        activeTransaction = stashedStartTransaction({
+          name: navigationEvent.url,
```

Comment:
Hmm there is no way for us to access a paramaterized url like `company/:companyid/people/:peopleid`? 

It's fine if there isn't, but I think that is a real concern for people.

Reply:
No, we only get a full resolved URL here :(